panes/add-person-to-page.vue

```diff
@@ -0,0 +1,155 @@
+      min-height: 53px;
+
+      .user-image {
+        flex: 0 0 32px;
+        height: 32px;
+        margin-right: 15px;
+        width: 32px;
+      }
+
+      .user-name,
+      .user-name-empty {
+        @include primary-text();
+
+        border: none;
+        flex: 1 0 auto;
+        line-height: 1.4;
+        padding-right: 0 15px;
+        // no left padding (it's handled by the image),
+        // so name displays to the left if image doesn't exist
+        text-align: left;
+      }
+
+      .user-name-empty {
+        @include tertiary-text();
+      }
+    }
+  }
+</style>
+
+<template>
+  <div class="add-person-to-page-pane">
+    <div class="add-person-to-page-pane-input">
+      <input
+        type="text"
+        class="add-person-to-page-pane-input-field"
+        placeholder="Search for someone"
+        ref="search"
+        v-model="query"
+        @keyup="onSearchKeyup">
+    </div>
+    <div class="add-person-to-page-pane-readout">
+      <ul class="add-person-to-page-pane-readout-list" ref="list">
+        <li v-for="(user, index) in users" class="add-person-to-page-pane-item" @click.stop.prevent="addPersonToPage(user)">
+          <img v-if="user.imageUrl" class="user-image" :src="user.imageUrl" />
+          <span v-if="user.name" class="user-name">{{ user.name }}</span>
+          <span v-else class="user-name-empty">{{ user.username }}</span>
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+  import _ from 'lodash';
+  import { postJSON } from '../lib/core-data/api';
+  import { searchRoute } from '../lib/utils/references';
+
+  function buildUserQuery(query) {
+    const str = _.isString(query) && query.toLowerCase() || '';
+
+    if (str !== '') {
+      // allow filtering by name / username (weighing the names more than usernames)
+      return {
+        multi_match: {
+          query: str,
+          fields: ['name^2', 'username']
+        }
+      };
+    } else {
+      // list all users
+      return {
+        match_all: {}
+      };
+    }
+  }
+
+  export default {
+    data() {
+      return {
+        query: '',
+        users: []
+      };
+    },
+    methods: {
+      fetchUsers() {
+        const query = this.query,
+          prefix = _.get(this.$store, 'state.site.prefix');
+
+        return postJSON(prefix + searchRoute, {
+          index: 'users',
+          type: 'general',
+          body: { query: buildUserQuery(query) }
+        }).then((res) => {
+          const hits = _.get(res, 'hits.hits') || [],
+            users = _.map(hits, (hit) => hit._source);
+
+          // set the filtered user data
+          this.users = users;
+        });
+      },
+      onSearchKeyup: _.debounce(function () {
+        this.fetchUsers();
+      }, 300),
+      addPersonToPage(user) {
+        this.$store.dispatch('startProgress', 'save');
+        return this.$store.dispatch('updatePageList', { user }).then(() => {
+          this.$store.dispatch('finishProgress', 'save');
```

Comment:
I'd recommend adding `finishProgress` and closing the pane and such to the `catch` for the promise as well, unless the error handling is already happening elsewhere

Reply:
Good point, I need to catch that if it fails for some reason